File: Searching/Ruby/jump_search.rb
def jump_search(arr, val, arrLength)
    step = Math.sqrt(arrLength).to_i
    prev = 0

    while arr[[step, arrLength].min] < val
        prev = step 
        step += Math.sqrt(arrLength).to_i
        if prev >= arrLength 
            return -1
        end
    end

    while arr[prev] < val 
        prev += 1 

        if prev == [step, arrLength].min 
            return -1
        end
    end

    if arr[prev] == val 
        return prev
    end

    return -1
end

arr = [ 0, 1, 1, 2, 3, 5, 8, 13, 21, 
    34, 55, 89, 144, 233, 377, 610 ]

p jump_search(arr, 55, arr.length)

File: Searching/Python/jump_search.py
import math 

def jumpsearch(arr, val, arrLength):

    step = math.sqrt(arrLength)

    prev = 0
    while arr[int(min(step, arrLength)-1)] < val:
        prev = step 
        step += math.sqrt(arrLength)
        if prev >= arrLength:
            return -1 

    while arr[int(prev)] < x:
        prev += 1

        if prev == min(step, arrLength):
            return -1

    if arr[int(prev)] == arrLength:
        return prev 

    return -1
        